nemo/collections/nlp/models/language_modeling/megatron_bert_model.py

```diff
@@ -862,3 +866,41 @@ def on_load_checkpoint(self, checkpoint) -> None:
                 parallel_state.set_virtual_pipeline_model_parallel_rank(i)
                 self.model[i].module.load_state_dict(checkpoint[f'model{i}'], strict=True)
             parallel_state.set_virtual_pipeline_model_parallel_rank(0)
+
+    def on_train_batch_end(self, outputs, dataloader_iter: Any, batch_idx: int, unused: Optional[int] = 0) -> None:
+        super().on_train_batch_end(outputs, dataloader_iter, batch_idx)
+
+        # TODO: Replace with newer override for scheduler.step() instead of
```

Comment:
why is this related to the new microbatch loader?

Reply:
This as well pulled from gpt3 code . @erhoo82  might know better.

Reply:
We use that method so we need to override it with the datloader_iter signature